zebra-state/src/service.rs

```diff
@@ -118,6 +136,17 @@ impl StateService {
         self.mem.any_chain_contains(hash) || &self.sled.finalized_tip_hash() == hash
     }
 
+    /// Returns true if the given hash has been committed to either the finalized
+    /// or non-finalized state.
+    fn contains(&self, block: &Block) -> bool {
```

Comment:
Tweak: the queue also contains blocks, so let's qualify this function name.

```suggestion
    fn contains_committed_block(&self, block: &Block) -> bool {
```